### ISSUE TITLE
Card.ReplaceEffect can crash the game/return error

### ISSUE THREAD

**Issue by edo9300** [Fluorohydride/ygopro-core#301]
When calling ReplaceEffect on a card that has registered an ``EFFECT_TYPE_FIELD+EFFECT_TYPE_TRIGGER_O``, the effect isn't properly cleared inside the core, and when it's raised, its handler will be a malformed pointer, resulting in: or the lua api catching this and returning ``"CallFunction": attempt to call an error function`` otherwise the core straight up crashing. Sample puzzle, here jackfrost's effect is being replaced with mystic tomato's, upon attack declaration with your mosnter, this situation will happen.

```
Debug.ReloadFieldBegin(DUEL_ATTACK_FIRST_TURN,5)
Debug.SetPlayerInfo(0,8000,0,0)
Debug.SetPlayerInfo(1,8000,0,0)

Debug.AddCard(74677422,0,0,LOCATION_MZONE,4,POS_FACEUP_ATTACK)
Debug.AddCard(61318483,1,1,LOCATION_MZONE,3,POS_FACEUP_DEFENSE):ReplaceEffect(83011277,0)

Debug.ReloadFieldEnd()
```

**Comment by DailyShana** [comment]
the effect is of `EFFECT_TYPE_FIELD+EFFECT_TYPE_TRIGGER_O` with range `LOCATION_HAND`.
When `Debug.AddCard`, hand trigger is always registered to the game_field(in `apply_field_effect`), even the handler is not in hand. When removing effect in replacing effect, as the handler is not in hand, the hand trigger is not removed from game field:
https://github.com/Fluorohydride/ygopro-core/blob/17d9dd15a8ae192f6f5d3a1a9800d16ddba72548/card.cpp#L1707-L1708

**Comment by edo9300** [comment]
I see, so the issue is with Debug.AddCard

**Comment by DailyShana** [comment]
fixed in 1b2b74aeb2ce52f7998231ed5af4db04d3d1d7c0
